targets/TARGET_STM/i2c_api.c

```diff
@@ -508,10 +634,15 @@ int i2c_stop(i2c_t *obj)
      *  re-init HAL state
      */
     if (obj_s->XferOperation != I2C_FIRST_AND_LAST_FRAME) {
-        i2c_init(obj, obj_s->sda, obj_s->scl);
-    }
 
-    return 0;
+#ifdef DEVICE_I2CSLAVE
+        const bool is_slave = obj_s->slave ? true : false;
+#else
+        const bool is_slave = false;
+#endif
+
+        i2c_init(obj, obj_s->sda, obj_s->scl, is_slave);
+    }
 }
 
 int i2c_byte_read(i2c_t *obj, int last)
```

Comment:
@scartmell-arm 
`i2c_byte_read` and  `i2c_byte_write` should be removed since it was removed from API